src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -418,23 +383,31 @@ class QueryAppsDataValidator {
     return true;
   }
 
-  bool FindMandatoryParameters(
+  bool CheckMandatoryParametersPresent(
       const smart_objects::SmartObject& app_data) const {
-    const bool kIsPackageNameNotExists =
-        app_data.keyExists(json::android) &&
-        !app_data[json::android].keyExists(json::packageName);
-    const bool kIsUrlSchemeNotExists =
-        app_data.keyExists(json::ios) &&
-        !app_data[json::ios].keyExists(json::urlScheme);
-
-    if (!app_data.keyExists(json::appId) || !app_data.keyExists(json::name) ||
-        kIsPackageNameNotExists || kIsUrlSchemeNotExists) {
+    if (app_data.keyExists(json::android) &&
+        !app_data[json::android].keyExists(json::packageName)) {
+      return false;
+    }
+
+    if (app_data.keyExists(json::ios) &&
+        !app_data[json::ios].keyExists(json::urlScheme)) {
       return false;
     }
+
+    if (!app_data.keyExists(json::appId)) {
+      return false;
+    }
+
+    if (!app_data.keyExists(json::name)) {
+      return false;
+    }
+
     return true;
   }
 
   smart_objects::SmartObject& data_;
+  std::set<std::string> applications_id_set_;
```

Comment:
@Kozoriz 
I would suggest to avoid of naming variable with its type since it can be changed later on.
It would be better something like application_ids